snapcraft/parts/parts.py

```diff
@@ -173,6 +173,7 @@ def run(
                 for action in actions:
                     message = _action_message(action)
                     emit.progress(f"Executing parts lifecycle: {message}")
+                    print(f"\033]2;Executing parts lifecycle: {message}\007")
```

Comment:
nice addition, but your original assessment seems correct, craft-cli is about the presentation. You can work with @tigarmo on figuring out what the best param for emit.progress can be